### PR TITLE
Add endpoints, update build and update tests

### PR DESCRIPTION
**Pull request by sqweelygig**
* Add endpoints for `contact`, `teammate` and `conversation.listRecent`
  * In this I have deliberately done all the ones I needed for SyncBot
  * In this I have deliberately not done any with pagination
  * I also did any that were in-between the two bounds above
* Added a simple test script to make encoding `FRONT_TEST_KEYS` easier on a terminal, and changed the interface to make compulsory details that are required to make tests pass
* Updated test scripts to reflect changes on the Front API and that Front free instances only have one teammate
* Updated the build target because Bluebird requires Iterable, which is not in es5

Change-Type: minor

### CODE REVIEW

File: lib/index.ts [hunk 1]
```diff
@@ -203,6 +225,12 @@ export class Front {
 		return httpServer;
 	}
 
+	// Utility method for occasions where we have the actual url, eg `_links`
+	public get(url: string, callback?: Callback<Object>): Promise<Object> {
```
Comment (hedss) [inline lib/index.ts]: Probably worth calling this `getUrl` or `getLinks` as `get` is a bit ambiguous.

File: test/duff.json [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+  "testChannel": "cha_blah",
+  "testConversationId": "cnv_blah",
+  "testCommentId": "com_blah",
+  "testInbox": "Blah",
+  "testTopicIssue": {
+    "owner": "blah",
+    "repo": "blah",
+    "issue": 123
+  }
+}
```
Comment (hedss) [inline test/duff.json]: Missing newline.

File: test/inbox.ts [hunk 3]
```diff
@@ -125,7 +125,7 @@ describe('Inboxes', function () {
 			response._links.should.exist;
 			response._links.self.should.startWith(`${apiUrl}/inboxes/${keys.testInboxId}/teammates`);
 			response._results.should.exist;
-			response._results.length.should.be.gt(1);
+			response._results.length.should.be.gte(1);
```
Comment (hedss) [inline test/inbox.ts:128]: Has something changed in the API that means this and line 112 don't hold true anymore?

File: tsconfig.json [hunk 4]
```diff
@@ -6,7 +6,7 @@
 		"preserveConstEnums": true,
 		"strictNullChecks": true,
 		"sourceMap": true,
-		"target": "es5",
+		"target": "es6",
```
Comment (hedss) [inline tsconfig.json]: Is there a reason we don't want to support <ES6?

**Review by hedss — COMMENTED** [review]
Looks good, couple of comments. We should ideally move to building this in Circle and publishing it from there too. Shall raise an issue.

File: lib/index.ts (hunk 1 above)
Comment (sqweelygig) [reply lib/index.ts]: ack, will do

File: test/duff.json (hunk 2 above)
Comment (sqweelygig) [reply test/duff.json]: ack, will do

File: test/inbox.ts (hunk 3 above)
Comment (sqweelygig) [reply test/inbox.ts:128]: My test (it being free) instance only has one teammate.

File: tsconfig.json (hunk 4 above)
Comment (sqweelygig) [reply tsconfig.json]: I get loads of Iterable not found errors in Bluebird when I target es5

File: test/duff.json (hunk 2 above)
Comment (sqweelygig) [reply test/duff.json]: Note to self: done

**Review by hedss — APPROVED** [review]
LGTM.

**Pull request merged**